classes/local/manager.php

```diff
@@ -196,20 +195,31 @@ static public function cloudfront_pem_exists() {
         if ('cf' !== $config->signingmethod) {
             return '';
         }
-        $path = $config->cloudfrontprivatekeypemfilepathname;
-        $fileformatted = true;
+        $path = $config->cloudfrontprivatekey;
         $text = 'settings:presignedcloudfronturl:cloudfront_pem_found';
         $type = 'notifysuccess';
-
-        $cert = file_get_contents($path);
-        if ((strpos($cert, '-----') !== 0) || openssl_pkey_get_private($cert) == false) {
-            $fileformatted = false;
-        }
-        $exits = file_exists($path) && is_readable($path) && $fileformatted;
-        if (false === $exits) {
+        if (false === self::parse_cloudfront_private_key($path)) {
             $text = 'settings:presignedcloudfronturl:cloudfront_pem_not_found';
             $type = 'notifyproblem';
         }
         return $OUTPUT->notification(get_string($text, OBJECTFS_PLUGIN_NAME), $type);
     }
+
+    /**
+     * @param string $cloudfrontprivatekey
+     * @return bool|resource
+     */
+    static public function parse_cloudfront_private_key($cloudfrontprivatekey) {
+        global $CFG;
+        $path = $CFG->dataroot . $cloudfrontprivatekey;
+        if (file_exists($path) && is_readable($path)) {
+            $cert = file_get_contents($path);
+            $parsedkey = openssl_pkey_get_private($cert);
```

Comment:
why not close the if here and have the same code path after this?

Reply:
done